### PR TITLE
Align elastic-agent-standalone manifest with the kubernetes package changes

### PR DESCRIPTION
**Pull request by tetianakravchenko**
Signed-off-by: Tetiana Kravchenko <tetiana.kravchenko@elastic.co>
## What does this PR do?
In kubernetes package were added some data-streams configuration adjustments. This PR backports those changes to elastic-agent standalone manifest.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues
- Closes https://github.com/elastic/beats/issues/29565

### CODE REVIEW

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml [hunk 1]
```diff
@@ -214,20 +224,75 @@ data:
...
           namespace: default
         streams:
           - data_stream:
-              dataset: generic
-            symlinks: true
+              dataset: kubernetes.container_logs
+              type: logs
+            prospector.scanner.symlinks: true
+            parsers:
+              - container:
```
Comment (MichaelKatsoulis) [inline deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml:241]: I believe we should keep the manifest as simple as possible and leave the defaults with not comments. The extra fields are documented for anyone who wants to change settings. I would just leave 
```
parsers:
        - container: ~
```

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml [hunk 2]
```diff
@@ -482,7 +547,7 @@ spec:
             - name: ES_PASSWORD
               value: ""
             - name: ES_HOST
-              value: ""
+              value: "elastic-package-stack_elasticsearch_1"
```
Comment (MichaelKatsoulis) [inline deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml:550]: Was this change intentional ? This is a proposed manifest and I don't think `elastic-package-stack_elasticsearch_1` is of any value to a customer.

Comment (tetianakravchenko) [reply deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml]: I've already removed it - it was used to test locally

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml (hunk 1 above)
Comment (tetianakravchenko) [reply deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml:241]: it is current default - see https://github.com/elastic/integrations/blob/master/packages/kubernetes/data_stream/container_logs/agent/stream/stream.yml.hbs#L7-L10, was introduced in https://github.com/elastic/integrations/pull/2345 to make it possible to configure those default values

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml [hunk 3]
```diff
@@ -54,6 +54,7 @@ data:
               - event
             period: 10s
             add_metadata: true
+            skip_older: true
```
Comment (tetianakravchenko) [inline deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml:57]: was introduced in https://github.com/elastic/integrations/pull/2342

File: deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml (hunk 1 above)
Comment (MichaelKatsoulis) [reply deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml:241]: Defaults are `all` streams and [auto](url) format. See in https://github.com/elastic/integrations/blob/master/packages/kubernetes/data_stream/container_logs/manifest.yml#L27.  
For filestream input if we don't set them  like 
```
- container: ~
```
it takes the [defaults](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-filestream.html#_container). 
For `additionalParsersConfig` we just added the possibility for a user to add another parser via advanced options.
 The default is still commented out.

Comment (tetianakravchenko) [reply deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml]: ah, ok, I understood what do you mean, I thought to set it the same way as done in the package, but you are right - in the end it is the same.
setting default implicitly - https://github.com/elastic/beats/pull/29595/commits/7c00ce97aa300be3570a36719b0dc60787a22339

**Review by ChrsMark — COMMENTED** [review]
Thanks for working on this! I left one small comment about the `skip_older` which is mostly to provide option for disabling this and hence we should better avoid exposing this with the default value.

**Review by MichaelKatsoulis — APPROVED** [review]

**Review by ChrsMark — APPROVED** [review]
🚀
